File: leetcode/1781.所有子字符串美丽值之和.go
func beautySum(s string) int {
    n := len(s)
    ans := 0
    for i := 0; i < n; i++ {
        mp := [26]int{}
        for j := i; j < n; j++ {
            mx := 0
            mi := n
            mp[s[j] - 'a'] += 1
            for _, x := range mp {
                if x != 0 {
                    mx = max(mx, x)
                    mi = min(mi, x)
                }
            }
            ans += mx - mi
        }
    }
    return ans
}

File: leetcode/1781.所有子字符串美丽值之和.py
class Solution:
    def beautySum(self, s: str) -> int:
        res = 0
        for i in range(len(s)):
            cnt = Counter()
            mx = 0
            for j in range(i, len(s)):
                cnt[s[j]] += 1
                mx = max(mx, cnt[s[j]])
                res += mx - min(cnt.values())
        return res